### DISCUSSION TITLE
laravel back pro - files field. "[Key] expected string length to be >= 1, but found string length of 0" error when i try to perform upload.

### DISCUSSION THREAD

**Discussion by prescriptionlifeline** [Laravel-Backpack/community-forum#1021] [Q&A (Help)]
I added the following field to my CRUD controller (the `setupUpdateOperation` method):

```php
        CRUD::field([
            'name' => 'image1',
            'label' => 'Image 1',
            'type' => 'image',
            'withFiles' => true,
            'crop' => true,
            'aspect_ratio' => 1.33333,
        ])->tab($board)
        ->withFiles([
            'disk' => 's3public',
            'path' => 'slideshow',
        ]);
```
Now, when I try to upload an image I took with my phone I get this error:

```
[2024-06-09 19:57:57] local.ERROR: InvalidArgumentException: Found 1 error while validating the input provided for the GetObject operation:
[Key] expected string length to be >= 1, but found string length of 0 in /var/www/website.com/vendor/aws/aws-sdk-php/src/Api/Validator.php:65
```

Here' what I have in my config/filesystems.php file:

```
        's3public' => [
            'driver' => 's3',
            'key' => env('AWS_ACCESS_KEY_ID'),
            'secret' => env('AWS_SECRET_ACCESS_KEY'),
            'region' => env('AWS_DEFAULT_REGION'),
            'bucket' => mycompany-public',
            'url' => env('AWS_URL'),
            'endpoint' => env('AWS_ENDPOINT'),
            'use_path_style_endpoint' => env('AWS_USE_PATH_STYLE_ENDPOINT', false),
	    //'throw' => false,
            'visibility' => 'public'
        ],
```

Any ideas?

**Comment by pxpm** [comment]
From the error it seems `env('AWS_ACCESS_KEY_ID'),` is returning an empty string, or not defined and return null. 

Ensure you have the proper key in your .env file. If you are using something to cache those values make sure you clear the framework cache with `php artisan optimize:clear`. 

You can always open a tinker session: `php artisan tinker`, and just write there: `env('AWS_ACCESS_KEY_ID')` and check the returned result. 

I am going to close this as I don't think this is an issue with Backpack. 

If I am wrong, and be sure sometimes I am, please re-open or comment back so that we can re-evaluate it. 🙏 

Cheers
